checker.py

```diff
@@ -0,0 +1,244 @@
+
+def check_missing(ref):
+    for hid in objs[ref]['extids']['handle']:
+        if hid not in handles:
+          return False
+    return True
+
+def get_shock_file(sid):
+    return '/disk1/shock/data/%s/%s/%s/%s/%s.data' % (sid[0:2], sid[2:4], sid[4:6], sid, sid)
+   
+def count_genes(fn):
+    ct =0
+    if not os.path.exists(fn):
+        print("Missing handle file " + fn)
+        return -1
+    with gzip.open(fn, 'rb') as f:
+        for li in f:
+            items=li.decode('utf-8').split('\t')
+            if len(items)>1 and items[2]=='gene':
+                ct+=1
+    return ct
+
+def read_kvs(ref):
+    kvs=dict()
+    for kv in objs[ref]['meta']:
+       kvs[kv['k']] = kv['v']
+    return kvs
+
+def check_genome(ref, odata):
+    if 'original_source_file_name' not in odata:
+        gff=handles[odata['gff_handle_ref']]
+        gct=count_genes(get_shock_file(gff['id']))
+        kvs = read_kvs(ref)
+        if int(kvs['Number of Protein Encoding Genes']) == gct:
+           return True
+        print(gct)
+        print(odata.keys())
+        print(odata['gff_handle_ref'])
+        print(ref)
+        return BAD
+    fn1=odata['original_source_file_name']
+    hrec = handles[odata['genbank_handle_ref']]
+    fn1=odata['original_source_file_name']
+    fn2=hrec['file_name'].replace('.gz','')
+    if fn1==fn2:
+        return GOOD
+    print(fn1)
+    print(fn2)
+    return BAD
+
+def check_cat(odata):
+     print(odata.keys())
+     print(odata['classifier_handle_ref'])
+     print(odata['lib_name'])
+     hid = odata['handle']['hid']
+     oid = odata['handle']['id']
+     id2=handles[hid]['id']
+     if oid==id2:
+         return GOOD
+     else:
+         return BAD
+def check_gff(odata):
+     hid = odata['handle']['hid']
+     oid = odata['handle']['id']
+     id2=handles[hid]['id']
+     if oid==id2:
+         return GOOD
+     else:
+         return BAD
+
+def check_report_type(odata):
+    good=0
+    l = odata['file_links']
+    l.extend(odata['html_links'])
+    #l.append(odata['direct_html'])
+    for f in l:
+        if f is None:
+           continue
+        if f=='':
+           continue
+        if f['handle'] in handles:
+           hid=f['handle']
+           id2=handles[hid]['id']
+           oid=f['URL'].split('/')[-1]
+           if oid==id2:
+              good+=1
+           else:
+              return BAD
+    return GOOD
+        
+def check_bowtie(odata):
+     hid = odata['handle']['hid']
+     oid = odata['handle']['id']
+     id2=handles[hid]['id']
+     if oid==id2:
+         return GOOD
+     else:
+         return BAD
+
+def check_single(odata):
+       hid = odata['lib']['file']['hid']
+       oid = odata['lib']['file']['id']
+       id2=handles[hid]['id']
+       if oid==id2:
+           return GOOD
+       else:
+           return BAD
+
+def check_paired2(odata):
+    good=0
+    print(odata)
+    li = [odata['handle_1']]
+    if 'handle_2' in odata:
+       li.append(odata['handle_2']['file'])
```

Comment:
This looks wrong. The first entry of the array will be a `handle_1` and the 2nd will be a `handle_2/file`

Reply:
I think this never showed up